deploy/helm/kubecf/templates/_config.tpl

```diff
@@ -0,0 +1,290 @@
+| _config.lookupManifest (list $.kubecf $path)
++-----------------------------------------------------------------------------------------
+| The same function as _config.lookup, except it searches $kubecf.manifest instead.
+==========================================================================================
+*/}}
+{{- define "_config.lookupManifest" }}
+  {{- $_ := include "_config.load" (first .) }}
+  {{- $kubecf := get (first .) "kubecf" }}
+  {{- $_ := set $kubecf "query" (join "." (rest .) | replace "/" ".") }}
+  {{- include "_config._lookup" (concat (list $kubecf $kubecf.manifest) (splitList "." $kubecf.query)) }}
+  {{- $_ := set $kubecf "required" false }}
+{{- end }}
+
+{{- /*
+==========================================================================================
+| _config._lookup (list $.kubecf $context $path)
++-----------------------------------------------------------------------------------------
+| Internal implementation of "_config.lookup" and "_config.lookupManifest".
+|
+| Lookup $path under $context and return either the value found, or the empty string.
+|
+| $context is either is an object (or array). It normally starts out as $.kubecf.config
+| or $.kubecf.manifest, but moves down the tree as _lookup calls itself recursively.
+|
+| $path is a list of properties to look up successively, e.g. ("stacks" $stack "releases").
+|
+| When the found value is `nil`, we still return the empty string and not a stringified
+| nil (which would be "<no data>"), because helm prints it as an empty string, but
+| otherwise treats as a non-empty string for all other purposes.
+|
+| $kubecf.retval is also set to the found entry in case the caller needs an object and
+| not just a string. This can also be used to disambiguate between nil and the empty
+| string (which can mean: not found).
+|
+| When $context is an array, then _lookup will look for an array element that has a "name"
+| property matching the first element of $path.
+|
+| Example: Looking for "instance_groups.api.jobs.cloud_controller_ng" in $.kubecf.manifest
+| is equivalent to the "/instance_groups/name=api/jobs/name=cloud_controller_ng" path in
+| JSON-patch.
+|
+| If the first element of $path for a list lookup contains a '=', the left side of specifies
+| the property to look for, in case it is not "name", e.g. "foo/os=linux/description".
++-----------------------------------------------------------------------------------------
+| Not implemented:
+|
+| When the full $path cannot be found, and the $.kubecf.required variable is true, then
+| _lookup should throw an error instead of returning an empty string. It could use
+| $.kubecf.query to provide a helpful error message. With the introduction of a
+| validation schema, this may no longer be required.
+==========================================================================================
+*/}}
+{{- define "_config._lookup" }}
+  {{- $kubecf := index . 0 }}
```

Comment:
No, here it is actually just `$.kubecf`. This is just the implementation function for both `_config.lookup` and `_config.lookupManifest`, it is not called anywhere else.

Should I rename it `_config._lookupImplementation`? The `_` was supposed to give a hint that this is internal to the file, but the `_` is kind of overloaded already, so maybe it is too subtle?

Reply:
The comments have changed a bit since `$.kubecf.config` was replaced by plain `$.Values` again, so I think this is now fine. Or is it still confusing.

It is full intentional that `lookup` and `_lookup` have different signaturs, and the docs show them correctly:

```
| _config.lookup (list $ $path)
| _config.lookupManifest (list $ $path)
| _config._lookup (list $.kubecf $context $path)
```